addons/stock/models/stock_move.py

```diff
@@ -883,6 +882,9 @@ def remove_prefix(text, prefix):
         for key in context:
             if key.startswith('default_'):
                 default_vals[remove_prefix(key, 'default_')] = context[key]
+        if 'company_id' in default_vals:
```

Comment:
You could add it upper. if key.startswith('default_') and key != 'default_company_id':